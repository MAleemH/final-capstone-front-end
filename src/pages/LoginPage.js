import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
// import { useDispatch } from 'react-redux';
// import { loginUser } from '../redux/User/userSlice';
import '../css/LoginPage.css';
import backImg from '../img/back.png';
import eyeImg from '../img/eye.png';
import fadingBreak from '../img/fading_break.png';
import logoImg from '../img/logo.png';

function LoginPage() {
  // const dispatch = useDispatch();
  const [passwordType, setPasswordType] = useState('password');
  const [email, setEmail] = useState('');
  const [inputPassword, setInputPassword] = useState('');
  const [errorMessage, setErrorMessage] = useState('');

  const handleLogin = async () => {
    const pass = inputPassword.replace(/\s/g, '').toLowerCase();
    console.log(email, pass);
    const userData = {
      email, password1: pass,
    };
    console.log(userData);
    // dispatch(loginUser(userData));
  };

  useEffect(() => {
    let active = true;
    (async () => {
      if (active) {
        setErrorMessage('');
      }
    })();
    return () => {
      active = false;
    };
  }, []);

  return (
    <div className="login_body">

      <header className="login_header">
        <nav>
          <Link to="/">
            <img src={backImg} alt="" />
          </Link>
        </nav>
      </header>

      <main className="login_main">
        <figure className="logo_fig">
          <Link to="/">
            <img src={logoImg} alt="" />
          </Link>
        </figure>

        <div className="fadebreak_box">
          <figure>
            {' '}
            <img src={fadingBreak} alt="" />
            {' '}
          </figure>
          <p>
            &nbsp;
            {`${errorMessage || 'Login with'}`}
            &nbsp;
            {' '}
          </p>
          <figure>
            {' '}
            <img className="rotate_breakline" src={fadingBreak} alt="" />
            {' '}
          </figure>
        </div>

        <form action="" className="login_form">
          <fieldset className="fieldset_border_none">
            <input className="input_name" type="email" placeholder="Email" aria-label="Input Email" onChange={(e) => setEmail(e.target.value)} required />

            <div className="password_box">
              <input className="input_password" type={passwordType} placeholder="Password" aria-label="Input Password" onChange={(e) => setInputPassword(e.target.value)} required />
              {passwordType === 'password'
                ? (
                  <figure className="eyebox">
                    <button className="password_display" type="button" onClick={() => setPasswordType('text')}>
                      <img src={eyeImg} alt="" />
                    </button>
                  </figure>
                )
                : (
                  <figure className="eyebox">
                    <button className="password_display" type="button" onClick={() => setPasswordType('password')}>
                      <img src={eyeImg} alt="" />
                    </button>
                  </figure>
                )}

            </div>
          </fieldset>

          <fieldset className="fieldset_border_none login_action">
            <button type="button" onClick={handleLogin}>Submit</button>
          </fieldset>
        </form>

        <div className="login_main_div">
          <p>
            {' '}
            <Link className="td_none" to="/register"> need an account? </Link>
            {' '}
          </p>

          <button type="button">
            <Link className="td_none" to="/forgot"> Forgot password </Link>
          </button>
        </div>

      </main>

      <footer className="login_footer">
        <ul className="list_style_none">
          <li>
            {' '}
            <Link className="td_none" to="/register"> terms </Link>
            {' '}
          </li>
          <li>
            {' '}
            <Link className="td_none" to="/register"> privacy </Link>
            {' '}
          </li>
        </ul>
      </footer>

    </div>
  );
}

export default LoginPage;
