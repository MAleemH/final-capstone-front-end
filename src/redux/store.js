import { configureStore } from '@reduxjs/toolkit';
import therapyReducer from './Therapy/therapySlice';
import userReducer from './User/userSlice';

const store = configureStore({
  reducer: {
    therapy: therapyReducer,
    user: userReducer,
  },
});

export default store;
