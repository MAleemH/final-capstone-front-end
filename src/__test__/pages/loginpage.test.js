import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import LoginPage from '../../pages/LoginPage';

it('LoginPage component renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <LoginPage />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
