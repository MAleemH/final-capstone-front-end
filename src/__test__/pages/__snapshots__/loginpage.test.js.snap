// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component renders correctly 1`] = `
<div
  className="login_body"
>
  <header
    className="login_header"
  >
    <nav>
      <button
        className="back_none"
        onClick={[Function]}
        type="button"
      >
        <img
          alt=""
          src="back.png"
        />
      </button>
    </nav>
  </header>
  <main
    className="login_main"
  >
    <figure
      className="logo_fig"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt=""
          src="logo.png"
        />
      </a>
    </figure>
    <div
      className="fadebreak_box"
    >
      <figure>
         
        <img
          alt=""
          src="fading_break.png"
        />
         
      </figure>
      <p>
         
        Login with
         
         
      </p>
      <figure>
         
        <img
          alt=""
          className="rotate_breakline"
          src="fading_break.png"
        />
         
      </figure>
    </div>
    <form
      action=""
      className="login_form"
    >
      <fieldset
        className="fieldset_border_none"
      >
        <input
          aria-label="Input Email"
          className="input_name"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
        />
        <div
          className="password_box"
        >
          <input
            aria-label="Input Password"
            className="input_password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
          />
          <figure
            className="eyebox"
          >
            <button
              className="password_display"
              onClick={[Function]}
              type="button"
            >
              <img
                alt=""
                src="eye.png"
              />
            </button>
          </figure>
        </div>
      </fieldset>
      <fieldset
        className="fieldset_border_none login_action"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Submit
        </button>
      </fieldset>
    </form>
    <div
      className="login_main_div"
    >
      <p>
         
        <a
          className="td_none"
          href="/register"
          onClick={[Function]}
        >
           need an account? 
        </a>
         
      </p>
      <button
        type="button"
      >
        <a
          className="td_none"
          href="/forgot"
          onClick={[Function]}
        >
           Forgot password 
        </a>
      </button>
    </div>
  </main>
  <footer
    className="login_footer"
  >
    <ul
      className="list_style_none"
    >
      <li>
         
        <a
          className="td_none"
          href="/register"
          onClick={[Function]}
        >
           terms 
        </a>
         
      </li>
      <li>
         
        <a
          className="td_none"
          href="/register"
          onClick={[Function]}
        >
           privacy 
        </a>
         
      </li>
    </ul>
  </footer>
</div>
`;
